swingset/src/main/java/com/nqadmin/swingset/SSComboBox.java

```diff
@@ -914,6 +1006,106 @@ public void setSelectedValue(final Integer _value) {
+			// things are set up as wanted
+			return;
+		}
+		try (Model.Remodel remodel = optionSwingModel.getRemodel()) {
+			// Need to add or remove the nullItem
+			// Since selection is based on item, not index, we only need
+			// to consider adjusting the selection if the current selection:
+			// A) null/-1 and nullItem is being added then select nullItem/0
+			// B) nullItem/0 and nullItem is being removed then select null/-1.
+			//
+			// The whole idea is that if null was selected before this
+			// change, want it seleted after the change.
+			//
+			// If listeners are enabled, may exceptions about using null
+			boolean selectNull = false;
+			boolean selectNone = false;
+			if (wantNull) {
+				selectNull = super.getSelectedIndex() == -1;
+				nullItem = remodel.createComboItem(null, "", null);
+				remodel.add(0, nullItem);
+			} else {
+				selectNone = super.getSelectedItem() == nullItem;
+				remodel.remove(nullItem);
+				nullItem = null;
+			}
+
+			// only manipulate listeners and selection if needed
+			if (selectNull || selectNone) {
+				final ActionListener[] listeners = unregisterAllActionListeners(this);
+				try {
+					if (selectNull) {
+						super.setSelectedItem(nullItem);
+					}
+					if (selectNone) {
+						super.setSelectedItem(null);
+					}
+				} finally {
+					registerAllActionListeners(this, listeners);
+				}
+			}
+		}
+	}
+
+	//
+	// unregisterAllActionListeners/registerAllActionListeners kudos glazed lists
+	//
+	/**
+	 * A convenience method to unregister and return all {@link ActionListener}s
+	 * currently installed on the given <code>comboBox</code>. This is the only
+	 * technique we can rely on to prevent the <code>comboBox</code> from
+	 * broadcasting {@link ActionEvent}s at inappropriate times.
+	 *
+	 * This method is the logical inverse of {@link #registerAllActionListeners}.
+	 */
+	private static ActionListener[] unregisterAllActionListeners(JComboBox<?> comboBox) {
+		final ActionListener[] listeners = comboBox.getActionListeners();
+		for (ActionListener listener : listeners) {
+			comboBox.removeActionListener(listener);
+		}
+
+		return listeners;
+	}
+
+	/**
+	 * A convenience method to register all of the given <code>listeners</code>
+	 * with the given <code>comboBox</code>.
+	 *
+	 * This method is the logical inverse of {@link #unregisterAllActionListeners}.
+	 */
+	private static void registerAllActionListeners(JComboBox<?> comboBox, ActionListener[] listeners) {
+		for (ActionListener listener : listeners) {
+			comboBox.addActionListener(listener);
+		}
+	}
+
+	/**
+	 * After this, make some adjustments.
+	 * {@inheritDoc }
+	 */
+	@Override
+	public void setAllowNull(boolean _allowNull) {
+		ssCommon.setAllowNull(_allowNull);
+		adjustForNullItem();
+	}
+
+	/**
+	 * Catch this to make some adjustments after a change in metadata.
+	 * {@inheritDoc }
+	 */
+	@Override
+	public void metadataChange() {
+		SSComponentInterface.super.metadataChange();
+		adjustForNullItem();
+	}
+
```

Comment:
Use this instead of bind(rs, col) plus setRowSet(rs)

Reply:
I know we already merged this, but do you recall what this comment was referring to?

Reply:
I'll adjust the comment so it indicates looking for change in nullability.